Code/Framework/AzCore/Tests/Asset/AssetManagerLoadingTests.cpp

```diff
@@ -3171,6 +3182,99 @@ namespace UnitTest
         EXPECT_TRUE(nestedDependentAsset.IsReady());
     }
 
+    TEST_F(AssetManagerClearAssetReferenceTests, ReloadTest)
+    {
+        // Regression test - there was a bug where rapid reloads could get stuck due to the owning container being invalidated
+        // Note that for this bug to occur, the loaded asset needs to have dependencies
+        // Order of events to reproduce:
```

Comment:
this is great, a lot of these tests are hard to figure out what's going on due to the complexity of the setup + test, this makes it easy to understand what's happening.